lib/liquid/tags/render.rb

```diff
@@ -38,13 +41,25 @@ def render_tag(context, output)
         parse_context: parse_context
       )
 
-      inner_context = context.new_isolated_subcontext
-      inner_context.template_name = template_name
-      inner_context.partial = true
-      @attributes.each do |key, value|
-        inner_context[key] = context.evaluate(value)
+      context_variable_name = @alias_name || template_name.split('/').last
+
+      variable = if @variable_name_expr
+        context.evaluate(@variable_name_expr)
+      else
+        context.find_variable(template_name, raise_on_not_found: false)
+      end
+
+      variable = [variable] unless variable.is_a?(Array)
```

Comment:
Protip: `variable = Array(variable)`

Reply:
That said we could also maybe turn the `.each` into a method, so that we can avoid making a one-item array and do something like instead:

```ruby
if variable_is_a?(Array) then variable.each(&:method_name) else method_name(variable)
```

Reply:
This doesn't seem to be fully equivalent `variable = Array(variable)`. I think its treating `nil` as `[]` however need it to be `[nil]` so it loops.

I was originally going down the method path but there are so many params in this scope that would need to pass in `context`, `template_name`, `output`, `var`, `partial`

Reply:
Good point. Let's add for that then.

Reply:
Not sure on the Rubyist view on this, but just used a lambda to group the functionality, gets around excessive params and needing to create a fake array.